api/V1/readme.md

```diff
@@ -0,0 +1,279 @@
+```protobuf
+message Status {
+    StatusCode code = 1;
+    string message = 2;
+}
+```
+
+Advantages:
+- Consistent error reporting
+- Rich error information
+- Easy to extend
+- Supports both technical and user-friendly messages
+
+Some error codes already listed in the draft, however the list might be not full and would need to be revisited with each particular usecase in mind
+
+Standardized Status Codes
+```protobuf
+// Common status codes used across all services
+enum StatusCode {
+  // Reserve 0 for unspecified
+  STATUS_UNSPECIFIED = 0;
+  // Success codes (1-99)
+  STATUS_OK = 1;
+  // Error codes (100+)
+  // ERROR_DATA_NOT_AVAILABLE is returned if data availability sampling gurantees was required for request
+  // and failed. Keep in mind that data availability sampling will be ensured for all blocks in
+  // sampling window till the requested height
+  ERROR_DATA_NOT_AVAILABLE = 100;
+  // Fraud was detected in the chain before requested block
+  ERROR_FRAUD_DETECTED = 101;
+  ERROR_TIMEOUT = 102;
+  // invalid input was submit as part of request. More information is provided in Status message field.
+  ERROR_INVALID_REQUEST = 103;
+  // unexpected error happened during handling the request
+  ERROR_INTERNAL = 104;
+  // method is not implemented by the server
+  ERROR_NOT_IMPLEMENTED = 105;
+}
+```
+
+## Service-Specific Decisions
+
+### BlobService
+
+#### 1. Data Submission Approach
+Two options were considered for blob submission:
+```protobuf
+message SubmitDataRequest {
+    bytes namespace = 1;
+    repeated bytes data = 2;
+    SubmitOptions options = 3;
+}
+
+message SubmitBlobRequest {
+  repeated Blob blobs = 1;
+  SubmitOptions options = 2;
+}
+```
+
+Chose raw data submission over pre-constructed blobs because:
+- Simpler for users - they don't need to understand blob internals
+- Reduces client-side complexity
+- Better separation between data and protocol details
+
+#### 2. Subscription Support
+Prevoiusly only one subscription endpoint existed to subscribe to all blobs from same namespace.
+```protobuf
+rpc SubscribeAll(SubscribeAllRequest) returns (stream SubscribeAllResponse);
+```
+Added new subscription endpoint `SubscribeBlobs`, which allows to subscribe for blobs by commitments:
+```protobuf
+rpc SubscribeBlobs(BlobsSubscribeRequest) returns (stream BlobsSubscribeResponse);
+```
+
+Motivations:
+- Essential for async submission workflows.It allows subscribing to blobs submitted in async mode
+- Enables event-driven architectures. 
+- Better for monitoring submitted data
+- Reduces polling overhead
+
+
+### ShareService
+Has support for requesting all common sets of shares:
+```protobuf
+// ShareService - handles data share operations
+service ShareService {
+  // Core share operations
+  rpc GetShare(GetShareRequest) returns (GetShareResponse);
+  rpc GetEDS(GetEDSRequest) returns (GetEDSResponse);
+
+  // Range operations - support both V0 (row roots) and V1 (data root) responses
```

Comment:
Both responses below return `V0` though?

Reply:
Ah, in the protos I see there is also `GetRangeResponseV1` etc. Maybe mention that we only list v0 here to keep the readme conscise.